### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by paulidale**
A couple of coverity fixes.

### CODE REVIEW

**Comment by t-j-h** [comment]
Need to update message from "covert" to "coverity"

**Comment by paulidale** [comment]
That was the autocorrect.  It happened a second time when I edited it.
It would have made a bad impression.

File: crypto/asn1/bio_ndef.c [hunk 1]
```diff
@@ -184,7 +184,7 @@ static int ndef_suffix(BIO *b, unsigned char **pbuf, int *plen, void *parg)
         return 0;
 
     derlen = ASN1_item_ndef_i2d(ndef_aux->val, NULL, ndef_aux->it);
-    if ((p = OPENSSL_malloc(derlen)) == NULL) {
+    if (derlen < 0 || (p = OPENSSL_malloc(derlen)) == NULL) {
         ASN1err(ASN1_F_NDEF_SUFFIX, ERR_R_MALLOC_FAILURE);
```
Comment (t-j-h) [inline crypto/asn1/bio_ndef.c]: That error message does not really match up. Is the only reason the call fail is for malloc reasons?

Comment (paulidale) [reply crypto/asn1/bio_ndef.c]: It's the most plausible reason but not the only one.
I'll address this in the morning.

I suspect that coverity (almost _covert_ again) has added some new rules about negative return values.

Comment (paulidale) [reply crypto/asn1/bio_ndef.c]: Changed to not produce an error message for this new case.
The underlying code sometimes adds an error message already.

**Comment by paulidale** [comment]
Merged to master, thanks.

**Pull request closed without merging**